# Generated by Django 4.2.5 on 2024-02-14 16:34

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('lawapp', '0004_request_accepted_request_rejected'),
    ]

    operations = [
        migrations.AlterField(
            model_name='request',
            name='role',
            field=models.CharField(choices=[('plaintiff', 'Plaintiff/Petitioner'), ('defendant', 'Defendant/Respondent')], max_length=10),
        ),
        migrations.CreateModel(
            name='Case',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('case_name', models.CharField(max_length=512)),
                ('case_number', models.CharField(max_length=255)),
                ('client_role', models.CharField(choices=[('plaintiff', 'Plaintiff/Petitioner'), ('defendant', 'Defendant/Respondent')], max_length=20)),
                ('case_description', models.TextField()),
                ('date_launched', models.DateTimeField(auto_now_add=True, null=True)),
                ('open_status', models.BooleanField(default=True)),
                ('advocate', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='created_cases', to=settings.AUTH_USER_MODEL)),
                ('client', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='assigned_cases', to=settings.AUTH_USER_MODEL)),
            ],
        ),
    ]
